gensim/models/ldamodel.py

```diff
@@ -1020,28 +1020,38 @@ def diff(self, other, distance="kullback_leibler", num_words=100, n_ann_terms=10
         if distance == "jaccard":
             d1, d2 = fst_topics, snd_topics
 
-        z = np.zeros((t1_size, t2_size))
-        for topic1 in range(t1_size):
-            for topic2 in range(t2_size):
-                z[topic1][topic2] = distance_func(d1[topic1], d2[topic2])
-
-        if normed:
-            if np.abs(np.max(z)) > 1e-8:
-                z /= np.max(z)
-
-        annotation = [[None] * t1_size for _ in range(t2_size)]
+        if diagonal:
+            t_size = min(t1_size, t2_size)
+            z = np.zeros(t_size)
```

Comment:
If the no. of topics is different between two models (diff will not be a square matrix), then what should be returned in the `diagonal` case? Either raise an error for num_topics to be same or just return the diff of identical topic no.s till the smaller value of num_topic of both model (latter one implemented right now)?